WordPressAuthenticator/GoogleSignIn/URL+GoogleSignIn.swift

```diff
@@ -2,26 +2,50 @@ import Foundation
 
 extension URL {
 
-    // TODO: This is incomplete
-    static func googleSignInAuthURL(clientId: String) throws -> URL {
-        let baseURL = "https://accounts.google.com/o/oauth2/v2/auth"
+    // It's acceptable to force-unwrap here because, for this call to fail we'd need a developer
+    // error, which we would catch because the unit tests would crash.
+    static var googleSignInBaseURL = URL(string: "https://accounts.google.com/o/oauth2/v2/auth")!
 
+    static func googleSignInAuthURL(clientId: String, pkce: ProofKeyForCodeExchange) throws -> URL {
         let queryItems = [
             ("client_id", clientId),
+            ("code_challenge", pkce.codeCallenge),
+            ("code_challenge_method", pkce.mode.method),
             ("redirect_uri", redirectURI(from: clientId)),
-            ("response_type", "code")
+            ("response_type", "code"),
+            // TODO: We might want to add some of these or them configurable
+            //
+            // The request we make with the SDK asks for:
+            //
+            // - email
+            // - profile
+            // - https://www.googleapis.com/auth/userinfo.email
+            // - https://www.googleapis.com/auth/userinfo.profile
+            // - openid
+            //
+            // See https://developers.google.com/identity/protocols/oauth2/scopes
+            ("scope", "https://www.googleapis.com/auth/userinfo.email")
         ].map { URLQueryItem(name: $0.0, value: $0.1) }
 
         if #available(iOS 16.0, *) {
-            return URL(string: baseURL)!.appending(queryItems: queryItems)
+            return googleSignInBaseURL.appending(queryItems: queryItems)
         } else {
-            var components = URLComponents(string: baseURL)!
+            let baseURL = googleSignInBaseURL
+            guard var components = URLComponents(url: baseURL, resolvingAgainstBaseURL: false) else {
+                throw URLError(
+                    .unsupportedURL,
+                    userInfo: [
+                        NSLocalizedDescriptionKey: "Could not create `URLComponents` instance from \(baseURL)"
+                    ]
+                )
+            }
```

Comment:
Do you think this is acceptable as a `URLError`, or should I update it to another `OAuthError` `case` for consistency?

I'm leaning toward moving it to `OAuthError`.

Reply:
That's a great observation! After all, what are the chances `URLComponents` will not initialize with a valid input URL?

Reply:
Addressed in 3824a52